prime-router/src/test/kotlin/fhirengine/translation/hl7/utils/FhirBundleUtilsTests.kt

```diff
@@ -83,36 +99,44 @@ class FhirBundleUtilsTests {
     @Test
     fun `test convert incompatible fhir types`() {
         var convertedValue = FhirBundleUtils.convertFhirType(BooleanType("true"), "boolean", "id")
-        assertThat(convertedValue).isInstanceOf(BooleanType::class)
+        assertThat(convertedValue!!).isInstanceOf(BooleanType::class)
 
         convertedValue = FhirBundleUtils.convertFhirType(StringType("testing"), "string", "boolean")
-        assertThat(convertedValue).isInstanceOf(StringType::class)
+        assertThat(convertedValue!!).isInstanceOf(StringType::class)
     }
 
     @Test
     fun `test convert invalid values`() {
-        assertFailure {
-            FhirBundleUtils.convertFhirType(
-                StringType("testing"),
-                "string",
-                "date"
-            )
-        }
+        val logger = spyk(this.logger)
+        var convertedValue = FhirBundleUtils.convertFhirType(
+            StringType("testing"),
+            "string",
+            "date",
+            logger
+        )
 
-        assertFailure {
-            FhirBundleUtils.convertFhirType(
-                StringType("testing"),
-                "string",
-                "dateTime"
-            )
-        }
+        assertThat(convertedValue!!).isInstanceOf(StringType::class)
```

Comment:
Because if there is no way to convert them we just return the value as a String. I could try to make this throw an error if you think that would be a better way of handling it.